src/main/java/com/amazon/opendistroforelasticsearch/security/DefaultObjectMapper.java

```diff
@@ -214,6 +217,19 @@ public T run() throws Exception {
             return AccessController.doPrivileged(new PrivilegedExceptionAction<T>() {
                 @Override
                 public T run() throws Exception {
+                    if (jt.getBindings().getTypeParameters().get(0).getRawClass() == InternalUserV6.class) {
```

Comment:
Do we want to check if this is for the migration from 6.x to 7.x or we are doing this even if is called when invoked from 6.x to 6.x_higher?

Reply:
Yes, we want to do this check for both 6.x and 7.x. Preventing empty password is the expected behavior for both 6.x and 7.x. This PR is to fix the empty password issue in migration for existing domains that have had empty passwords. We're preventing empty password in Internal User creation in another PR.